### ISSUE TITLE
false negative in ruleset validation

### ISSUE THREAD

**Issue by J0WI** [EFForg/https-everywhere#5034]
Tests passed with ruleset validation failed. This result in a broken build for any further PR.
https://travis-ci.org/EFForg/https-everywhere/jobs/133667658#L247

cc @Hainish

**Comment by fuglede** [comment]
Is this one still relevant? Looks like builds are passing.

**Comment by J0WI** [comment]
This specific build should fail, not pass.

**Comment by Hainish** [comment]
I don't understand the problem - it looks like the ruleset validation passed, not failed as you indicated above.  What's the problem?

**Comment by J0WI** [comment]
Tests passed with the following error in the regex:
https://github.com/EFForg/https-everywhere/commit/551a513f5a3b14153e45ce48d58ef6a94947dc87#diff-b45950db5e9f08c8c1a6cb038a0290cdR197

```
failure: /opt/utils/../src/chrome/content/rules/Washington-Post-Company.xml failed test: The 'from' rule contains unescaped period in regular expression.  Try escaping it with a backslash.

ERROR: Validation of rulesets failed.
```
...
```
The command "docker run --rm -ti -v $(pwd):/opt -e FIREFOX=/$FIREFOX/firefox/firefox httpse bash test.sh" exited with 0.
```

**Comment by Hainish** [comment]
I've replicated the <rule> from the commit you mention above.  With docker I get the following:

```shell
(.venv)user@https-everywhere ~/workspace/https-everywhere (wapo) $ sudo docker run --rm -ti -v $(pwd):/opt -e FIREFOX=/$FIREFOX/firefox/firefox httpse bash -c "./test.sh"
+ '[' -n '' ']'
+ git rev-parse
++ git rev-parse --show-toplevel
+ cd /opt
+ ./test/rules.sh
+ '[' -n '' ']'
+ git rev-parse
++ git rev-parse --show-toplevel
+ cd /opt
+ source utils/mktemp.sh
+++ which mktemp
++ MKTEMP=/bin/mktemp
++ mktemp
++ /bin/mktemp
+ TMP=/tmp/tmp.pQ9n9JGyDg
+ trap 'rm "$TMP"' EXIT
+ '[' -d test/rules ']'
+ '[' 0 -gt 0 ']'
+ python2.7 test/rules/src/https_everywhere_checker/check_rules.py test/rules/coverage.checker.config
+ exit 0
+ rm /tmp/tmp.pQ9n9JGyDg
+ ./test/firefox.sh
+ '[' -n '' ']'
+ git rev-parse
++ git rev-parse --show-toplevel
+ cd /opt
+ source utils/mktemp.sh
+++ which mktemp
++ MKTEMP=/bin/mktemp
+ TEST_ADDON_PATH=./test/firefox/
++ mktemp -d
++ /bin/mktemp -d
+ PROFILE_DIRECTORY=/tmp/tmp.hWjuBvnz3m
+ trap 'rm -r "$PROFILE_DIRECTORY"' EXIT
+ HTTPSE_INSTALL_DIRECTORY=/tmp/tmp.hWjuBvnz3m/extensions/https-everywhere-eff@eff.org
+ ./makexpi.sh
Generating ruleset DB
failure: /opt/utils/../src/chrome/content/rules/Washington-Post-Company.xml failed test: The 'from' rule contains unescaped period in regular expression.  Try escaping it with a backslash.
ERROR: Validation of rulesets failed.
+ rm -r /tmp/tmp.hWjuBvnz3m
```

This is expected.

The cause of the pass that you saw in https://travis-ci.org/EFForg/https-everywhere/jobs/133667658#L247 was fixed by 0afdfdb5b26e855e510509931ae41eedb724332b
